npm/src/main.ts

```diff
@@ -96,9 +96,19 @@ export function getDebugServiceWorker(): IDebugServiceWorker {
   return proxy;
 }
 
-export function getLanguageService(): ILanguageService {
+export function getLanguageService(
+  readFile: (uri: string) => Promise<string | null> = () =>
+    Promise.resolve(null),
```

Comment:
I see you declare these defaults in a bunch of places - how about moving the default arguments to the `QSharpLanguageService` constructor (which ends up being the common code path), declaring the arguments as optional (i.e. using `?` in the argument list) here, and just omitting them in `worker-browser` and `worker-node`? That way you'll only end up defining the defaults in one place.